addons/api/addon/serializers/storage-bucket.js

```diff
@@ -12,27 +12,49 @@ const fieldsByCredentialType = {
     'secret_access_key',
     'region',
     'disable_credential_rotation',
+    'endpoint_url',
   ],
   dynamic: [
     'role_arn',
     'role_external_id',
     'role_session_name',
     'role_tags',
-    'disable_credential_rotation',
     'region',
+    'disable_credential_rotation',
+    'endpoint_url',
   ],
 };
 export default class StorageBucketSerializer extends ApplicationSerializer {
   serialize(snapshot) {
-    const serialized = super.serialize(...arguments);
     const { credentialType } = snapshot.record;
+    const pluginType = snapshot.record.plugin?.name;
+    const serialized = super.serialize(...arguments);
+
+    // Common for all pluginTypes
     if (!snapshot.record.isNew) {
       delete serialized.bucket_name;
       delete serialized.bucket_prefix;
     }
 
-    if (credentialType === TYPE_CREDENTIAL_DYNAMIC) {
-      serialized.attributes.disable_credential_rotation = true;
+    switch (pluginType) {
+      case 'minio':
+        return this.serializeMinio(serialized);
```

Comment:
I think inversing this might be a better approach. We can make MinIO the default option since it doesn't have any serialization, and we don't need a separate case for this. Thoughts?

Reply:
@DhariniJeeva 
I think you are totally right, ATM there is no need for specific MinIO serialization. So will perform a specific one just for AWS and return a default one valid for MinIO.
Thanks for the catch!